### ISSUE TITLE
[no-unused-properties] produces false-positive when destructured value is aliased

### ISSUE THREAD

**Issue by gajus** [sindresorhus/eslint-plugin-unicorn#2051]
```ts
const {
  project,
  triggers,
  abortSignal: userAbortSignal,
  debounce: userDebounce,
  onReady,
}: Configuration = {
  // as far as I can tell, this is a bug in unicorn/no-unused-properties
  // eslint-disable-next-line unicorn/no-unused-properties
  debounce: {
    wait: 1_000,
  },
  ...configurationInput,
};
```

Here `debounce` is reported as unused despite it being aliased to `userDebounce`, which is used in the the code.

https://github.com/gajus/turbowatch/commit/7a7451ee9b82f82b60c9e89392ab4a3bd57d1f6c
